### PR TITLE
[16.0] [MIG] stock_picking_return_restricted_qty: Migration to 16.0

### PR DESCRIPTION
**Pull request by ced-adhoc**
In this migration from v15 to v16, the following changes have been introduced:

- Operation Type Configuration: 
A new restrict_return_qty Boolean field has been added to stock.picking.type, allowing users to enable or disable the restriction on returning more products than delivered for each operation type.

- Updated Logic for Return Picking:
The logic in stock_picking_return.py now checks the restrict_return_qty field from the related picking type. The restriction will trigger the UserError only if this setting is enabled.

- Backward Compatibility:
If the setting is not enabled, the standard behavior is maintained. This behavior is widely used in purchases, where items are returned to the vendor together in one operation, regardless of the original delivery.

Steps to use:
1. Enable the "Restrict Return Quantity" option in Inventory > Configuration > Operation Types.
2. Create a picking, validate it, and test the return process with the restriction enabled.

### CODE REVIEW

**Comment by ced-adhoc** [comment]
Hi @rousseldenis, could you please check this PR? thanks in advance!

**Comment by mav-adhoc** [comment]
Techincal review OK!

**Review by jcadhoc — APPROVED** [review]

**Review by florenciafrigieri2 — APPROVED** [review]
Functional test ok. It's working even if you do 2 or more returns over the same picking.

**Comment by florenciafrigieri2** [comment]
@simahawk Hi Simone! I saw you in the mainteiners list. Can you please take a look at this PR? Thanks!

**Comment by ced-adhoc** [comment]
Hi @jbaudoux  ! could you please check this PR? Thanks in advance!

**Review by jcadhoc — APPROVED** [review]

**Review by mav-adhoc — APPROVED** [review]

**Comment by ced-adhoc** [comment]
Hi @pedrobaeza, could you please check this PR? Thanks in advance!

**Comment by pedrobaeza** [comment]
/ocabot migration stock_picking_return_restricted_qty

@sergio-teruel please confirm this migration is correct.

**Review by sergio-teruel — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@ced-adhoc @sergio-teruel In fact, as this module changes the standard behavior, the restriction should be enabled through a parameter.

**Comment by ced-adhoc** [comment]
@rousseldenis 
Hi Denis,
Thanks for the feedback on the migration. I saw the suggestion about adding a parameter to enable the restriction, but I’m a bit unclear on what you meant. Could you explain a bit more about it? 
Thanks in advance!

**Review by valentinagiuliani-rapsodoo — APPROVED** [review]

**Comment by rousseldenis** [comment]
> @rousseldenis Hi Denis, Thanks for the feedback on the migration. I saw the suggestion about adding a parameter to enable the restriction, but I’m a bit unclear on what you meant. Could you explain a bit more about it? Thanks in advance!

@ced-adhoc 

See this: https://github.com/OCA/stock-logistics-workflow/actions/runs/10812154210/job/29993153238#step:8:558

As you change the standard behavior, some modules are not aware of that restriction.

To avoid such problems, it is recommended to add a configuration parameter (res.config.settings/company) to enable the constraint (and launch the UserError).

Could you add it in a further commit (and rebase - to see the error here) ?

File: stock_picking_return_restricted_qty/wizard/stock_picking_return.py [hunk 1]
```diff
@@ -17,12 +17,17 @@ def _prepare_stock_return_picking_line_vals_from_move(self, stock_move):
         return val
 
     def _create_returns(self):
+        restrict_return_qty = self.picking_id.picking_type_id.restrict_return_qty
```
Comment (rousseldenis) [inline stock_picking_return_restricted_qty/wizard/stock_picking_return.py:20]: You did it well. In fact, just the parameter on picking type level should be sufficient.

Could you adapt CONFIGURATION fragment to reflect this change ? Thanks

File: stock_picking_return_restricted_qty/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (rousseldenis) [inline stock_picking_return_restricted_qty/views/res_config_settings_views.xml]: You can remove this

**Comment by mav-adhoc** [comment]
LGTM

**Comment by ced-adhoc** [comment]
Hi @rousseldenis ,
I've completed all the requested modifications. Please let me know if anything else is needed.

**Comment by ced-adhoc** [comment]
Hi @rousseldenis ,
Could you please check this? Thanks!

File: stock_picking_return_restricted_qty/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+To use this module, you need to:
+
+#. Create an outgoing or an internal picking.
+#. Include one or more products with enough stock.
+#. Validate the picking.
+#. Attemp to return more products than the delivered and see the warning message.
```
Comment (jbaudoux) [inline stock_picking_return_restricted_qty/readme/USAGE.rst]: Try was fine in this context :)
otherwise, it should be `Attempt` with `t` at the end

**Review by jbaudoux — COMMENTED** [review]
Can you split the migration commit into a pure migration commit and another for the improvements?

**Comment by ced-adhoc** [comment]
@jbaudoux all done! I guess

**Comment by jbaudoux** [comment]
> @jbaudoux all done! I guess

@ced-adhoc Nearly, the changes in wizard are in the wrong commit ;)

**Comment by ced-adhoc** [comment]
@jbaudoux thanks! done

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**